lib/mock_suey/type_checks/sorbet.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+gem "sorbet-runtime", "~> 0.5"
+require "sorbet-runtime"
+require "set"
+require "pathname"
+
+require_relative "../sorbet_rspec"
+require "mock_suey/ext/instance_class"
+
+module MockSuey
+  module TypeChecks
+    using Ext::InstanceClass
+
+    class Sorbet
+      RAISE_ON_MISSING_MESSAGE = "Please, set raise_on_missing_types to false to disable this error. Details: https://github.com/test-prof/mock-suey#raise_on_missing_types"
+
+      def initialize(load_dirs: [])
+        @load_dirs = Array(load_dirs)
+      end
+
+      def typecheck!(method_call, raise_on_missing: false)
```

Comment:
There's a room for improvement: the variable `method_call` is passed too many times.
Therefore, it should be better to extract these functions into a class or a structure `MethodCheck`, where `method_call` (and `mocked_obj`, `arguments`, `method_name`, etc...) will be available as instance variables.